sdk/rust/nym-sdk/examples/tcp_proxy_single_connection.rs

```diff
@@ -0,0 +1,177 @@
+use bincode;
+use nym_sdk::tcp_proxy;
+use rand::rngs::SmallRng;
+use rand::{Rng, SeedableRng};
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::sync::Arc;
+use tokio::io::AsyncWriteExt;
+use tokio::net::{TcpListener, TcpStream};
+use tokio::signal;
+use tokio::sync::Mutex;
+use tokio::task;
+use tokio_stream::StreamExt;
+use tokio_util::codec;
+use tracing_subscriber;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ExampleMessage {
+    message_id: i8,
+    message_bytes: Vec<u8>,
+}
+
+// This is a basic example which opens a single TCP connection and writes a bunch of messages between a client and an echo
+// server, so only uses a single session under the hood and doesn't really show off the message ordering capabilities; this is mainly
+// just a quick introductory illustration on how:
+// - the mixnet does message ordering
+// - the NymProxyClient and NymProxyServer can be hooked into and used to communicate between two otherwise pretty vanilla TcpStreams
+//
+// For a more irl example checkout tcp_proxy_multistream.rs
+#[tokio::main]
+async fn main() {
+    // Keep track of sent/received messages
+    let counter = Arc::new(Mutex::new(0));
```

Comment:
You could use an [AtomicU8](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU8.html), or similar here.